profiler/src/ProfilerEngine/Datadog.Profiler.Native/ApplicationStore.cpp

```diff
@@ -3,14 +3,57 @@
 
 #include "ApplicationStore.h"
 
+#include "ApplicationInfo.h"
 #include "IConfiguration.h"
 
 ApplicationStore::ApplicationStore(IConfiguration* configuration) :
     _pConfiguration{configuration}
 {
 }
 
-const std::string& ApplicationStore::GetName(std::string_view runtimeId)
+ApplicationInfo ApplicationStore::GetApplicationInfo(const std::string& runtimeId)
 {
-    return _pConfiguration->GetServiceName();
+    {
+        std::lock_guard lock(_infosLock);
+
+        const auto info = _infos.find(runtimeId);
+
+        if (info != _infos.end())
+        {
+            return info->second;
+        }
+    }
+
+    return
+    {
+        _pConfiguration->GetServiceName(),
+        _pConfiguration->GetEnvironment(),
+        _pConfiguration->GetVersion()
+    };
+}
+
+
+void ApplicationStore::SetApplicationInfo(const std::string runtimeId, const std::string serviceName, const std::string environment, const std::string version)
```

Comment:
seems that `const` on parameters are not part of the declaration in the header. (and the interface)

Reply:
Yeah, Resharper suggests to remove the const keyword on header/interface. I guess it means it doesn't change anything for the caller? I don't mind adding it back if you prefer

Reply:
you can remove it, my point was mainly to be consistent across the files